Mage.Sets/src/mage/cards/d/DesertWarfare.java

```diff
@@ -0,0 +1,200 @@
+
+        // Whenever you sacrifice a Desert and whenever a Desert card is put into your
+        // graveyard from your hand or library, put that card onto the battlefield under your control at the beginning of your next end step.
+
+        //Based on Seraph, Yuma, Proud Protector
+        Effect effect = new CreateDelayedTriggeredAbilityEffect((DelayedTriggeredAbility) new AtTheBeginOfNextEndStepDelayedTriggeredAbility(new DesertWarfareReturnEffect(), TargetController.YOU).setTriggerPhrase("Put that card onto the battlefield under your control at the beginning of your next end step.").setRuleVisible(false));
+
+        this.addAbility(new SacrificePermanentTriggeredAbility(Zone.BATTLEFIELD, effect, filter, TargetController.YOU, SetTargetPointer.PERMANENT, false).setTriggerPhrase("Whenever you sacrifice a Desert and whenever a Desert card is put into your graveyard from your hand or library, put that card onto the battlefield under your control at the beginning of your next end step.").setRuleVisible(false));
+
+        this.addAbility(new PutCardIntoGraveFromHandLibraryAllTriggeredAbility(Zone.BATTLEFIELD, effect,
+                false, TargetController.YOU, SetTargetPointer.CARD).setTriggerPhrase("Whenever you sacrifice a Desert and whenever a Desert card is put into your graveyard from your hand or library, put that card onto the battlefield under your control at the beginning of your next end step."));
+
+        //At the beginning of combat on your turn, if you control five or more Deserts, create that many 1/1 red, green, and white Sand Warrior creature tokens. They gain haste.
+        //Based on Palani's Hatcher
+        this.addAbility(new ConditionalInterveningIfTriggeredAbility(
+                new BeginningOfCombatTriggeredAbility(new DesertWarfareCreateTokensEffect(), TargetController.YOU, false),
+                new PermanentsOnTheBattlefieldCondition(filter, ComparisonType.OR_GREATER, 5, true),
+                "At the beginning of combat on your turn, if you control five or more Deserts, create that many 1/1 red, green, and white Sand Warrior creature tokens. They gain haste.").addHint(DesertsYouControlHint.instance));
+
+    }
+
+    private DesertWarfare(final DesertWarfare card) {
+        super(card);
+    }
+
+    @Override
+    public DesertWarfare copy() {
+        return new DesertWarfare(this);
+    }
+
+}
+
+//Based on PutCardIntoGraveFromAnywhereAllTriggeredAbility
+class PutCardIntoGraveFromHandLibraryAllTriggeredAbility extends TriggeredAbilityImpl {
+
+    private static FilterCard filterCard = new FilterCard();
+
+    static {
+        filterCard.add(SubType.DESERT.getPredicate());
+    }
+
+    private final SetTargetPointer setTargetPointer;
+
+    public PutCardIntoGraveFromHandLibraryAllTriggeredAbility(Zone zone, Effect effect, boolean optional, TargetController targetController, SetTargetPointer setTargetPointer) {
+        super(zone, effect, optional);
+        FilterCard filter = filterCard.copy();
```

Comment:
FilterOwnedCard exists.

This is a custom class, so it should include the card name, and params should just be hardcoded.